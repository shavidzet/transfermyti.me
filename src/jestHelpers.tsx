/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
