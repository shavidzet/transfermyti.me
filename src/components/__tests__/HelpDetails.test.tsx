import React from "react";
import { render, RenderResult } from "~/jestHelpers";
import HelpDetails from "../HelpDetails";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    title: "Test Title",
    children: "Test Contents",
    ...propOverrides,
  };

  const wrapper = render(<HelpDetails {...props} />);

  return { props, wrapper };
};

describe("the <HelpDetails> component", () => {
  test("matches its snapshot with valid props", () => {
    const { wrapper } = setup();

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });

  test("uses a default title if props.title is undefined", () => {
    const { wrapper } = setup({ title: undefined });

    expect(wrapper.getByText("Show/Hide Help")).toBeInTheDocument();
  });
});
