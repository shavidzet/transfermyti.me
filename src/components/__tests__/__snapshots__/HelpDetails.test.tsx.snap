// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <HelpDetails> component matches its snapshot with valid props 1`] = `
.emotion-1 {
  margin-bottom: 1rem;
  width: 100%;
  color: rgb(30,120,161);
}

.emotion-1 div,
.emotion-1 summary {
  margin: 0;
  padding: 0.5rem 0.25rem;
}

.emotion-1 p:not(first-of-type) {
  margin-top: 0;
}

.emotion-1 p:last-of-type {
  margin-bottom: 0;
}

.emotion-0 {
  font-weight: 600;
}

<details
  class="emotion-1"
>
  <summary
    class="emotion-0"
  >
    Test Title
  </summary>
  <div>
    Test Contents
  </div>
  <hr />
</details>
`;
